### PR TITLE
[IOTDB-5620] Fix flush stuck when there is a lot of time partitions in each DataRegion

### PR DESCRIPTION
**Pull request by HTHou**
## Description

When set Max_heap_size=2G and insert data of 7 years. The flush thread will be stuck. 

There is a simple way to reproduce. 

Set max_heap_size of data node to 200M, set write_memory_proportion to 10000000:1.

Then start standalone iotdb and execute the following sqls:

```sql
insert into root.sg.d1(time,s1) values(1,2);
flush;
insert into root.sg.d1(time,s1) values(2,2);
insert into root.sg.d1(time,s1) values(604800001,2);
flush;
```
Finally, you will see the last flush cannot execute successfully.

## Reason

The `evictOldPartition` method in `TimePartitionManager` always peek the first TimePartitionInfo in the TreeSet in the while loop. However, the memSize of the TimePartitionInfo may be 0. In this case, the loop won't be ended.

## Solution

Change the `treeSet.first()` to `treeSet.pollFirst()`.

### CODE REVIEW

File: integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBInsertMultiPartitionIT.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.sql.Connection;
+import java.sql.Statement;
+
+import static org.junit.Assert.fail;
+
+@RunWith(IoTDBTestRunner.class)
+@Category({LocalStandaloneIT.class, ClusterIT.class, RemoteIT.class})
```
Comment (JackieTien97) [inline integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBInsertMultiPartitionIT.java:40]:
```suggestion
@Category({LocalStandaloneIT.class, ClusterIT.class})
```

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java [hunk 2]
```diff
@@ -1606,13 +1606,14 @@ private void initStorageEngineAllocate(Properties properties) {
 
     String allocationRatioForWrite = properties.getProperty("write_memory_proportion", "19:1");
     proportions = allocationRatioForWrite.split(":");
-    int proportionForMemTable = Integer.parseInt(proportions[0].trim());
+    int proportionForMemTable = Integer.parseInt(proportions[0].replace("\\", "").trim());
     int proportionForTimePartitionInfo = Integer.parseInt(proportions[1].trim());
+    logger.info("proportionForTimePartitionInfo = {}", proportionForTimePartitionInfo);
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:1611]:
```suggestion
```

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java [hunk 3]
```diff
@@ -1606,13 +1606,14 @@ private void initStorageEngineAllocate(Properties properties) {
 
     String allocationRatioForWrite = properties.getProperty("write_memory_proportion", "19:1");
     proportions = allocationRatioForWrite.split(":");
-    int proportionForMemTable = Integer.parseInt(proportions[0].trim());
+    int proportionForMemTable = Integer.parseInt(proportions[0].replace("\\", "").trim());
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:1609]: replace("\\", "") is for what?

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java (hunk 3 above)
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:1609]: image

The serialize method of property will add `\` for `:`.

File: integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBInsertMultiPartitionIT.java (hunk 1 above)
Comment (HTHou) [reply integration-test/src/test/java/org/apache/iotdb/db/it/IoTDBInsertMultiPartitionIT.java:40]: `RemoteIT.class` is also in other IT cases. It is never used in actions.

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java [hunk 4]
```diff
@@ -1606,13 +1606,13 @@ private void initStorageEngineAllocate(Properties properties) {
...
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:1609]:
```suggestion
    int proportionForMemTable = Integer.parseInt(proportions[0].trim());
```

**Pull request merged**